pump/server.go

```diff
@@ -266,6 +268,11 @@ func (s *Server) PullBinlogs(in *binlog.PullBinlogReq, stream binlog.Pump_PullBi
 	// don't use pos.Suffix now, use offset like last commitTS
 	last := in.StartFrom.Offset
 
+	gcTS := atomic.LoadInt64(&s.gcTS)
+	if last < gcTS {
+		log.Error("drainer request a purged binlog TS, some binlog events may be loss", zap.Int64("gc TS", gcTS), zap.Reflect("request", in))
+	}
```

Comment:
`Append` already has a `gcTS` field that keeps track of the same information, it seems it would be much simpler to do this check in `Append.PullCommitBinlog`.

Reply:
no, I want to do it in `server`, because I want to know where is the request from, even though the request doesn't have this information yet.

Reply:
Maybe we can add a `GetLastGCTS` method to `Storage`. That way we can access it here without maintaining yet another copy of the same piece of information in `Server`.

Reply:
no, its's obvious that the specific time of gc is controlled outside, and it is only implemented. now we need write like this because of our bad gc implement base on leveldb

Reply:
you should handle it in Storage let the user know the the the binlog is purged(maybe change PullCommitBinlog api to return error) not just check at first request,  currently, it will have some undefined behavior if  you gc a point before the point that drainer is fetching.